transfer/multichain_queue.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2021 The "MysteriumNetwork/payments" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package transfer
+
+import (
+	"errors"
+)
+
+// MultichainQueue is map of Queues which are able to
+// queue transaction execution funcs in parallel for each chain.
+type MultichainQueue struct {
+	qs map[int64](*Queue)
+}
+
+// ErrMissingChainId is returned when a given chainID doesnt have an existing queue .
+var ErrMissingChainId = errors.New("no queue for chain")
+
+// NewMultichainQueue returns a new empty multichain queue.
+func NewMultichainQueue() *MultichainQueue {
+	return &MultichainQueue{
+		qs: make(map[int64]*Queue),
+	}
+}
+
+// AddQueue will add a queue with the given chain ID.
+func (mq *MultichainQueue) AddQueue(chainID int64, queue *Queue) {
+	mq.qs[chainID] = queue
+}
+
+// TransactionEnqueue will enqueue a given transaction in a given chain and respond on the given resp channel.
```

Comment:
Well honestly at this point the current enqueue method will only really find a queue and queue a transaction. Dont want to leave such comments in case someone updates he original queue and then this comment becomes a lie.